src/serializer.ts

```diff
@@ -17,14 +19,157 @@ interface PaymentJSON {
  * Provides functionality to serialize from protocol buffers to JSON objects.
  */
 class Serializer {
+/**
+   * Convert a Transaction to a JSON representation.
+   *
+   * @param {proto.Transaction} transaction A Transaction to convert.
+   * @returns {Object} The Transaction as JSON.
+   */
+  public static transactionToJSON(
+    transaction: Transaction
+  ): object | undefined {
+    // Serialize the protocol buffer to a JSON representation.
+    var object: any = transaction.toObject();
+
+    // Convert fields names where direct conversion is possible.
+    this.convertPropertyName("sequence", "Sequence", object);
+    this.convertPropertyName("signingPublicKey", "SigningPubKey", object);
+    this.convertPropertyName(
+      "lastLedgerSequence",
+      "LastLedgerSequence",
+      object
+    );
+
+    // Delete unsupported fields from the protocol buffer.
+    delete object.memosList;
+    delete object.flags;
+    delete object.signature;
+    delete object.signersList;
+    delete object.sourceTag;
+    delete object.accountTransactionId;
+
+    // Encode SigningPubKey to hex, which is what ripple-binary-codec expects.
+    object.SigningPubKey = Utils.toHex(transaction.getSigningPublicKey_asU8());
+
+    // Convert account field, handling X-Addresses if needed.
+    const accountAddress = transaction.getAccount();
+    if (!accountAddress) {
+      return undefined;
```

Comment:
You shouldn't need to `return undefined`. Just a `return` will do (unless Typescript or a linter is yelling about it).

Reply:
As a non-JavaScript programmer, I find this syntax really hard. `return` implies the function has no return type in most other languages.

I've made the change since I trust you to be more syntactically / idiomatic minded than myself.